### PR TITLE
Add profiling to integration service

### PR DESCRIPTION
**Pull request by 14rcole**
Enabling profiling in our controller will allow us to start gathering data on how we can improve the performance of the integration service.

See [this blog](https://eng.d2iq.com/blog/profiling-kubernetes-controllers-with-pprof/) to understand how to gather logs from pprof once it is running in the cluster.  I've tested this on quickcluster and I can get the logs, but we'll get a better performance picture from real-world workloads.

## Maintainers will complete the following section

- [ ] Commit messages are descriptive enough ([hints](https://www.freecodecamp.org/news/how-to-write-better-git-commit-messages/))
- [ ] Code coverage from testing does not decrease and new code is covered ([check the PR coverage on codecov](https://app.codecov.io/gh/redhat-appstudio/integration-service/pulls))
- [ ] [Controllers diagrams](https://github.com/redhat-appstudio/integration-service/tree/main/docs) are updated when PR changes controllers code  (if applicable)

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -102,6 +103,7 @@ func main() {
 				},
 			},
 		}),
+		PprofBindAddress: "127.0.0.1:8081",
```
[the remark this answers is not in the document]
Comment (MartinBasti) [reply main.go:106]: Isn't profiler itself causing performance degradation by doing profiling?

[the remark this answers is not in the document]
Comment (MartinBasti) [reply main.go:106]: It looks that go can do magic and it's profiler is prod safe, with only 0.5% performance degradation
https://medium.com/google-cloud/continuous-profiling-of-go-programs-96d4416af77b

[the remark this answers is not in the document]
Comment (14rcole) [reply main.go:106]: That's great news.  If you'd still rather not run profiling in production I can add a conditional to only set it up in stage.

**Comment by 14rcole** [comment]
e2e tests failure is because another service is already using port 8081.  Going to update to port 9091.  This doesn't impact functionality, just the port you have to query to get the profiling statistics.

File: main.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (14rcole) [reply main.go]: I read that article a bit more closely.

> We target an additional 5% overhead for CPU and heap allocation profiling

That 0.5% was giving an example for if you have 10 replicas of a pod.  If you run the profiler 1/10 as often or only run it on one pod that's 0.5% overall.  That said, it looks like the performance is highly configurable.  Regardless, I would lean toward only enabling this in stage for now.  I'll see if there's a way to determine what environment we're running in when the controller manager is created.

[the remark this answers is not in the document]
Comment (14rcole) [reply main.go]: Step 1: https://github.com/redhat-appstudio/infra-deployments/pull/2905

[the remark this answers is not in the document]
Comment (MartinBasti) [reply main.go]: Enabling it on demand would give me better sleep than running by default in prod, thanks.

[the remark this answers is not in the document]
Comment (MartinBasti) [reply main.go]: We can also use envvars, it should be easier to manage than configmap.

**Comment by 14rcole** [comment]
Closing this for the time being.  Will open a new PR if adding profiling becomes a priority.

**Pull request closed without merging**